backend/tracim_backend/lib/search/search.py

```diff
@@ -357,20 +378,97 @@ def index_content(self, content: ContentInContext) -> None:
-        indexed_content.save(using=self.es, index=self.index_document_alias)
+        if self._can_index_content(content):
+            file_ = content.get_b64_file()
+            if file_:
+                indexed_content.b64_file = file_
+                indexed_content.save(
+                    using=self.es,
+                    pipeline="attachment",
+                    index=self.index_document_alias,
+                    request_timeout=self._config.SEARCH__ELASTICSEARCH__REQUEST_TIMEOUT,
+                )
+                return
+            logger.debug(
+                self,
+                'Content file of content "{}" will be not indexed: no content'.format(
+                    content.content_id
+                ),
+            )
+        indexed_content.save(
+            using=self.es,
+            index=self.index_document_alias,
+            request_timeout=self._config.SEARCH__ELASTICSEARCH__REQUEST_TIMEOUT,
+        )
+
+    def _can_index_content(self, content: ContentInContext) -> bool:
+        if not self._config.SEARCH__ELASTICSEARCH__USE_INGEST:
+            logger.debug(
+                self,
+                'Content file of content "{}" will be not indexed: ingest mode disabled'.format(
+                    content.content_id
+                ),
+            )
+            return False
+
+        if not content.content.depot_file or content.size is None:
+            logger.debug(
+                self,
+                'Content file of content "{}" will be not indexed:  not a correct file format'.format(
+                    content.content_id
+                ),
+            )
+            return False
+
+        # INFO - G.M - 2019-06-24 - check mimetype validity
+        if (
+            self._config.SEARCH__ELASTICSEARCH__INGEST__MIMETYPE_WHITELIST
+            and content.mimetype
+            not in self._config.SEARCH__ELASTICSEARCH__INGEST__MIMETYPE_WHITELIST
+        ):
+            logger.debug(
+                self,
+                'Content file of content "{}" will be not indexed: mimetype "{}" not allowed to be indexed (whitelist rule)'.format(
+                    content.content_id, content.mimetype
+                ),
+            )
+            return False
+
+        # INFO - G.M - 2019-06-24 - check mimetype validity
+        if (
+            self._config.SEARCH__ELASTICSEARCH__INGEST__MIMETYPE_BLACKLIST
+            and content.mimetype in self._config.SEARCH__ELASTICSEARCH__INGEST__MIMETYPE_BLACKLIST
+        ):
+            logger.debug(
+                self,
+                'Content file of content "{}" will be not indexed: mimetype "{}" not allowed to be indexed (blacklist rule)'.format(
+                    content.content_id, content.mimetype
+                ),
+            )
+            return False
+
+        if content.size == 0:
+            logger.debug(
+                self,
+                'Content file of content "{}" will be not indexed:  empty'.format(
```

Comment:
=> 'Skip binary indexation of content "{}": empty file'

Reply:
done.